### ISSUE TITLE
Bug: gridcell ARIA role incorrectly added to Day button element

### ISSUE THREAD

**Issue by gpbl** [gpbl/react-day-picker#1688]
### Discussed in https://github.com/gpbl/react-day-picker/discussions/1674

<div type='discussions-op-text'>

<sup>Originally posted by **rihardsgravis** February  2, 2023</sup>
### Bug description

The ARIA `gridcell` role is added to the Day button, but that is an incorrect role for the button since it does not represent a cell in a grid – it is in fact a child of a cell. Accessibility tests using the component are failing.

### Expected behaviour

The role for the button could be `button` or omitted at all since it's a HTM  button component and the role for it is clear.
</div>

**Comment by jacobsickels** [comment]
Waiting on a fix for this!

**Comment by gpbl** [comment]
Reason this is taking so long is because it is a breaking change: changing the HTML structure will likely break tests selectors and styles.

I could finally make some progress with this in PR #2089 and will cut a mayor release for this fix. More details coming soon... thanks a lot for your patience!
